src/test/java/book_store/service/ShoppingCartServiceImplTest.java

```diff
@@ -0,0 +1,200 @@
+package book_store.service;
+
+import book_store.dto.cartItemDto.UpdateCartItemRequest;
+import book_store.dto.shoppingCart.AddToCartRequest;
+import book_store.dto.shoppingCart.ShoppingCartDto;
+import book_store.exception.EntityNotFoundException;
+import book_store.mapper.BookMapper;
+import book_store.mapper.ShoppingCartMapper;
+import book_store.model.Book;
+import book_store.model.CartItem;
+import book_store.model.ShoppingCart;
+import book_store.model.User;
+import book_store.repository.cartItem.CartItemRepository;
+import book_store.repository.shoppingCart.ShoppingCartRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import java.util.List;
+import java.util.Optional;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class ShoppingCartServiceImplTest {
+    @Mock
+    private ShoppingCartRepository shoppingCartRepository;
+
+    @Mock
+    private CartItemRepository cartItemRepository;
+
+    @Mock
+    private ShoppingCartMapper shoppingCartMapper;
+
+    @InjectMocks
+    private ShoppingCartServiceImpl shoppingCartService;
+
+    @Mock
+    private BookMapper bookMapper;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    @DisplayName("Test findUserShoppingCart: Success - User Exists, ShoppingCart Returned")
+    void testFindUserShoppingCart_Success_UserExists_ShoppingCartReturned() {
```

Comment:
test name invalid, should be in pattern like:
`<methodUnderTest>_<state>_<expectedBehavior>`